### PR TITLE
Added examples

### PR DESCRIPTION
**Pull request by alexkwolfe**
This PR adds the addtional optional `examples` property to fields. The example strings are expanded to parsed type objects. The expanded examples allow for using the `raw` property for posting instructions, and the `normal` property for "internal" examples.

### CODE REVIEW

**Comment by alexkwolfe** [comment]
Hey @cgrayson, do you mind looking this over?

File: fields/standard.json [hunk 1]
```diff
@@ -1,47 +1,46 @@
...
+{"id":"gender","name":"Gender","description":"Gender","type":"gender","examples":["female", "F", "Male", "m", "O", "other"]}
 {"id":"industry","name":"Industry","description":"Company industry","type":"string"}
 {"id":"ip_address","name":"IP Address","description":"IP Address","type":"ip"}
 {"id":"is_military","name":"Military","description":"Does the buyer have current or previous military service?","type":"boolean"}
 {"id":"landing_page","name":"Landing Page","description":"The landing page","type":"string"}
 {"id":"last_name","type":"last_name","name":"Last Name","description":"Last name"}
-{"id":"leadid_id", "name": "LeadiD ID", "type":"string","description":"LeadiD identifier"}
-{"id":"marital_status","name":"Marital Status","description":"Marital status: married, not married, separated","type":"string"}
+{"id":"leadid_id", "name": "LeadiD ID", "type":"string","description":"LeadiD identifier","examples":["0C1D258E-0775-21A7-0014-1BB6444A5431"]}
+{"id":"marital_status","name":"Marital Status","description":"The consumers current marital status","type":"string","examples":["married","not married","separated"]}
```
Comment (cgrayson) [inline fields/standard.json]: Apostrophe: `"The consumer's…"`

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "prepublish": "cake build"
   },
   "dependencies": {
-    "leadconduit-types": "^3.2.1",
+    "leadconduit-types": "^4.0.0",
```
Comment (cgrayson) [inline package.json]: Curious why you changed this here, instead of doing it with `cut-release` after merge.

File: src/index.coffee [hunk 3]
```diff
@@ -7,6 +7,17 @@ types = require 'leadconduit-types'
 
 seedDir = path.join(__dirname, '..', 'fields')
 
+expandExamples = (field) ->
+  return field unless field.type
+  return field unless field.examples?.length
+  field.examples = field.examples.map (example) ->
+    return example if field.type == 'string'
+    str = (example?.raw or example)?.trim()
```
Comment (cgrayson) [inline src/index.coffee]: Are either of these `?`s necessary?

File: spec/seed-spec.coffee [hunk 4]
```diff
@@ -19,8 +20,19 @@ describe 'Seeds', ->
...
+      {normal: 0.3, valid: true, raw: '$0.30'},
+      {normal: 1, valid: true, raw: '1'},
+      {normal: 10.5, valid: true, raw: '10.50'},
+      {normal: 12, valid: true, raw: '12'}
+    ]
+
+  it 'should have valid examples', ->
+    for field in fields
+      for example in field.examples or []
+        assert example.valid, "#{field.id} has an invalid example: #{example}" if _.isPlainObject(example)
```
Comment (cgrayson) [inline spec/seed-spec.coffee]: This emits `AssertionError: [whatever] has an invalid example: [object Object]`.

File: spec/example-spec.coffee [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      country_code: 'US'
+      type: null
+      valid: true
+    ]
+
+  it 'should re-expand expanded examples', ->
+    field =
+      id: 'tellie'
+      type: 'phone'
+      examples: ['512-788-1111']
```
Comment (cgrayson) [inline spec/example-spec.coffee]: Might factor `field` out into a `beforeEach()`.

**Comment by cgrayson** [comment]
Otherwise :+1:

File: package.json (hunk 2 above)
Comment (alexkwolfe) [inline package.json]: This is a dependency version.

File: src/index.coffee (hunk 3 above)
Comment (alexkwolfe) [inline src/index.coffee]: They aren't strictly necessary. They just protect against the case where somehow a `null` got in the list of examples.

Comment (alexkwolfe) [inline src/index.coffee]: Given that, I should probably compact the array.

**Pull request merged**